endpoints/setuid.go

```diff
@@ -25,8 +26,14 @@ const (
 	chromeiOSStrLen = len(chromeiOSStr)
 )
 
-func NewSetUIDEndpoint(cfg config.HostCookie, perms gdpr.Permissions, pbsanalytics analytics.PBSAnalyticsModule, metrics pbsmetrics.MetricsEngine) httprouter.Handle {
+func NewSetUIDEndpoint(cfg config.HostCookie, syncers map[openrtb_ext.BidderName]usersync.Usersyncer, perms gdpr.Permissions, pbsanalytics analytics.PBSAnalyticsModule, metrics pbsmetrics.MetricsEngine) httprouter.Handle {
 	cookieTTL := time.Duration(cfg.TTL) * 24 * time.Hour
+
+	validFamilyNameMap := make(map[string]struct{})
```

Comment:
The other methods which need syncer information take the entire map. I thought it best to keep the signature for this endpoint similar and pull out an optimized name hashset. The hashset creation only runs once on startup.

Reply:
You could have the value of the map be the `FamilyName`, so that down below you will have it handy for metrics.

Reply:
Family Name to Bidder Name is not a 1:1 mapping.

Reply:
Nope, usersyncer doesn't have it .... there is an old confusion in the code, `FamilyName` is supposed to be the bidder name, after resolving aliases. But usersync has given it its own definition ... since 90% of the time it is the same. We would have to refactor `Usersyncer` to be able to access this information.